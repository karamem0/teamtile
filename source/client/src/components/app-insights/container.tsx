//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import { ReactPlugin, withAITracking } from '@microsoft/applicationinsights-react-js';
import { ApplicationInsights } from '@microsoft/applicationinsights-web';

import { createBrowserHistory } from 'history';

import Presenter from './presenter';

const browserHistory = createBrowserHistory({ basename: '' });

const reactPlugin = new ReactPlugin();

const appInsights = new ApplicationInsights({
  config: {
    instrumentationKey: process.env.APP_INSIGHTS_INSTRUMENTATION_KEY,
    extensions: [ reactPlugin ],
    extensionConfig: {
      [reactPlugin.identifier]: { history: browserHistory }
    }
  }
});

appInsights.loadAppInsights();

export default withAITracking(reactPlugin, Presenter);
