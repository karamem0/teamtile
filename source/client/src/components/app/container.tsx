//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { PublicClientApplication } from '@azure/msal-browser';

import { authConfig } from '../../config/auth-config';

import Presenter from './presenter';

export default function App (): React.ReactElement | null {

  return (
    <Presenter instance={new PublicClientApplication(authConfig)} />
  );

}
