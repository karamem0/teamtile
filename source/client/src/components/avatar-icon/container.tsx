//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { SizeValue } from '@fluentui/react-northstar';

import { Icon } from '../../types/entity';

import Presenter from './presenter';

interface AvatarIconProps {
  icon: Icon | undefined,
  name: string | null | undefined,
  size: SizeValue | undefined
}

export default function AvatarIcon ({
  icon,
  name,
  size
}: AvatarIconProps): React.ReactElement | null {

  return (
    <Presenter
      icon={icon}
      name={name}
      size={size} />
  );

}
