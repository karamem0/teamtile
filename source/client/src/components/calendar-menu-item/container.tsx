//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { app, appInstallDialog, HostClientType } from '@microsoft/teams-js';

import { useServiceContext } from '../../contexts/service-context';
import { TeamWithMail } from '../../types/entity';
import { ItemKey, ItemValue } from '../../types/state';

import Presenter from './presenter';

interface CalendarMenuItemProps {
  itemKey: ItemKey,
  itemValue: ItemValue
}

export default function CalendarMenuItem ({
  itemValue
}: CalendarMenuItemProps): React.ReactElement | null {

  const { services } = useServiceContext();
  const [ loading, setLoading ] = React.useState<boolean>(false);

  const handleClick = React.useCallback(async (_, data: TeamWithMail | undefined) => {
    if (!data?.id || !data?.internalId || !data.mail) {
      return;
    }
    const [ name, domain ] = data.mail.split('@');
    const context = await app.getContext();
    switch (context.app.host.clientType) {
      case HostClientType.desktop:
      case HostClientType.web: {
        try {
          setLoading(true);
          const appId = process.env.APP_CHANNEL_CALENDAR_APP_ID;
          const tab = await services
            .getTabs(data.id, data.internalId)
            .then((tabs) => tabs.find((tab) => tab.appId === appId));
          if (tab?.webUrl) {
            app.openLink(tab.webUrl);
          } else if (appInstallDialog.isSupported()) {
            await appInstallDialog.openAppInstallDialog({ appId });
          } else {
            const [ name, domain ] = data.mail.split('@');
            app.openLink(`https://outlook.office.com/calendar/group/${domain}/${name}`);
          }
        } finally {
          setLoading(false);
        }
        break;
      }
      default:
        app.openLink(`https://outlook.office.com/calendar/group/${domain}/${name}`);
        break;
    }
  }, [
    services
  ]);

  if (!itemValue.drive) {
    return null;
  }

  return (
    <Presenter
      loading={loading}
      team={itemValue}
      onClick={handleClick} />
  );

}
