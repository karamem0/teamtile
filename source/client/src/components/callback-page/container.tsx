//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { app } from '@microsoft/teams-js';

import { useMsal } from '@azure/msal-react';

import Presenter from './presenter';

export default function CallbackPage (): React.ReactElement | null {

  const { instance } = useMsal();

  React.useEffect(() => {
    (async () => {
      await app.initialize();
      await instance.handleRedirectPromise();
    })();
  }, [
    instance
  ]);

  return (
    <Presenter />
  );

}
