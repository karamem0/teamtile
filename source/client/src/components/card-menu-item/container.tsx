//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { EventHandler } from '../../types/common';

import Presenter from './presenter';

interface CardMenuItemProps {
  icon: React.ReactNode | undefined,
  content: React.ReactNode | undefined,
  onClick?: EventHandler | undefined
}

export default function CardMenuItem ({
  icon,
  content,
  onClick
}: CardMenuItemProps): React.ReactElement | null {

  return (
    <Presenter
      content={content}
      icon={icon}
      onClick={onClick} />
  );

}
