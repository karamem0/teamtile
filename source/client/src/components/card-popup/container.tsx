//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';
import { useDebounce } from 'react-use';

import { InputProps, PopupProps } from '@fluentui/react-northstar';

import { EventHandler } from '../../types/common';

import Presenter from './presenter';

interface CardPopupProps<T> {
  items: T[],
  predicate: (filter: string, item: T) => boolean,
  renderer: (items: T[]) => React.ReactNode,
  trigger: React.ReactElement,
  onOpenChange?: EventHandler<PopupProps>
}

export default function CardPopup <T> ({
  items,
  renderer,
  predicate,
  trigger,
  onOpenChange
}: CardPopupProps<T>): React.ReactElement | null {

  const [ filterItems, setFilterItems ] = React.useState(items);
  const [ filter, setFilter ] = React.useState<string | null>(null);

  useDebounce(() => {
    if (!filter) {
      setFilterItems(items);
    } else {
      setFilterItems(items.filter((item) => predicate(filter, item)));
    }
  }, 500, [
    items,
    filter
  ]);

  const handleInputChange = React.useCallback((_, data?: InputProps & { value: string } | undefined) => {
    setFilter(data?.value || null);
  }, []);

  return (
    <Presenter
      items={filterItems}
      renderer={renderer}
      trigger={trigger}
      onInputChange={handleInputChange}
      onOpenChange={onOpenChange} />
  );

}
