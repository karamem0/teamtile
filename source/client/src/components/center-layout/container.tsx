//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import Presenter from './presenter';

interface CenterLayoutProps {
  children: React.ReactNode
}

export default function CenterLayout ({
  children
}: CenterLayoutProps): React.ReactElement | null {

  return (
    <Presenter>
      {children}
    </Presenter>
  );

}
