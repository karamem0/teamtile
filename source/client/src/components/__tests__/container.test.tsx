//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

jest.mock('../home-page', () => ({
  HomePage: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="HomePage">
      {children}
    </div>
  )
}));

jest.mock('../main-page', () => ({
  MainPage: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="MainPage">
      {children}
    </div>
  )
}));

const errorContextValue = {
  setError: jest.fn()
};
jest.mock('../../contexts/error-context', () => ({
  useErrorContext: jest.fn().mockReturnValue(errorContextValue)
}));

jest.mock('../../contexts/reducer-context', () => ({
  ReducerContextProvider: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="ReducerContextProvider">
      {children}
    </div>
  )
}));

jest.mock('../../contexts/service-context', () => ({
  ServiceContextProvider: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="ServiceContextProvider">
      {children}
    </div>
  )
}));

const inTeamsValue = {
  inTeams: null as boolean | null
};
jest.mock('../../hooks/use-in-teams', () => ({
  useInTeams: jest.fn().mockReturnValue(inTeamsValue)
}));

import React from 'react';

import { render, screen } from '@testing-library/react';

import { Container } from '../container';

beforeEach(() => {
  jest.clearAllMocks();
  jest.restoreAllMocks();
});

describe('Container', () => {

  it('create shapshot if app is in teams', () => {
    inTeamsValue.inTeams = true;
    render(<Container />);
    expect(screen.queryAllByText(/^.*$/)[0]).toMatchSnapshot();
  });

  it('create shapshot if app is not in teams', () => {
    inTeamsValue.inTeams = false;
    render(<Container />);
    expect(screen.queryAllByText(/^.*$/)[0]).toMatchSnapshot();
  });

});
