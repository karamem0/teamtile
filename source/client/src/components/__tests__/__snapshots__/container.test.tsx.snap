// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container create shapshot if app is in teams 1`] = `
@media (max-width: 599px) {
  .emotion-0 {
    padding: 0.5rem;
  }
}

@media (min-width: 600px) {
  .emotion-0 {
    padding: 1rem;
  }
}

<body>
  <div>
    <div
      class="emotion-0"
    >
      <div
        data-testid="ReducerContextProvider"
      >
        <div
          data-testid="ServiceContextProvider"
        >
          <div
            data-testid="MainPage"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Container create shapshot if app is not in teams 1`] = `
<body>
  <div>
    <div
      data-testid="HomePage"
    />
  </div>
</body>
`;
